### PR TITLE
fix: [CDS-76756]: Show container step group compatible steps when step template is used inside container step group

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
### Summary

Issue: With the introduction of container step groups, there was update in steps which were shown in step palette when step group added was containerised.
This update was not done in use template api payload due to which stage dependent step templates were shown while user was trying add in containerised step group.

Fix: When use template is clicked from container step group , stepsV2 api call is made with updated payload which returns required response and that is used for getTemplate api further.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/98508399/97f77d86-a9c5-4d88-9388-2c75086113f9

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by rameshx — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]
Awesome. 
Can we write test case for this one?

**Comment by sonaligoyal-harness** [comment]
I will write it in separate PR.
This is CFD and  opened from long . So need to close this today.

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/hooks/useAddStepTemplate.ts [hunk 1]
```diff
@@ -176,5 +176,44 @@ export function useAddStepTemplate(props: AddStepTemplate): AddStepTemplateRetur
...
+        }
+      })
+        .then((response: ResponseStepCategory) => {
+          const stepCategoriesData = response?.data?.stepCategories
+          if (stepCategoriesData) {
+            const updatedStepTypes = getStepTypesFromCategories(stepCategoriesData)
+            resolve(updatedStepTypes)
+          } else resolve([])
+        })
+        .catch(_err => {
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/hooks/useAddStepTemplate.ts:200]: Do we need not to show the error from here?

Comment (sonaligoyal-harness) [reply src/modules/70-pipeline/hooks/useAddStepTemplate.ts:200]: No,

**Review by Eric761 — APPROVED** [review]

**Pull request merged**